src/Uri.php

```diff
@@ -316,7 +317,7 @@ public function withHost($host) : UriInterface
         }
 
         $new = clone $this;
-        $new->host = strtolower($host);
+        $new->host = mb_strtolower($host);
```

Comment:
Another option is to do feature detection, and use `mb_strtolower()` only if it's present on the system, and otherwise use `strtolower()`.

In most cases, if you are working with URIs that might contain multi-byte characters, the rest of your application likely does as well, so you'll have mbstring installed. If you don't, you'll hit the bug - and the recommendation will be "install mbstring".

Thoughts?

Reply:
@weierophinney I was just discussing that option with @geerteltink and he is worried about the performance.

Maybe we should add the check in bugfix version and in minor add the requirement?

Reply:
We can cache the first call to test - just have an internal property, and a method that caches the lookup. These sorts of lookups are _very_ inexpensive though.